swap_meet/vendor.py

```diff
@@ -1,2 +1,96 @@
+from queue import PriorityQueue
+from swap_meet.item import Item
+
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory  = None):
+        if not inventory:
+            inventory = []
+        self.inventory = inventory
+        
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self,item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        else:
+            item not in self.inventory
+            return False
+    
+    def get_by_category(self,category):
+        category_list= []
+        for item in self.inventory:
+            if item.category == category:
+                category_list.append(item)
+        return category_list
+    
+    def swap_items(self, other_vendor, my_item,their_item):
+        if my_item not in self.inventory or their_item not in other_vendor.inventory:
+            return False
+        if my_item in self.inventory:
```

Comment:
It's not hurting anything to check this, but it's executing an additional unnecessary loop, because we've already checked for whether `my_item` is _not_ in `self.inventory`

Reply:
I'm curious to know how I would refactor this without that if statement. Should I just go into removing the item?